extensions/resteasy-reactive/rest-client-reactive/deployment/src/main/java/io/quarkus/rest/client/reactive/deployment/RestClientReactiveProcessor.java

```diff
@@ -307,21 +308,14 @@ void addRestClientBeans(Capabilities capabilities,
                         //      return ((InterfaceClass)this.getDelegate()).get();
                         // }
                         MethodCreator methodCreator = classCreator.getMethodCreator(MethodDescriptor.of(method));
+                        methodCreator.setSignature(JandexUtil.reconstructGenericSignature(method));
 
                         // copy method annotations, there can be interceptors bound to them:
                         for (AnnotationInstance annotation : method.annotations()) {
                             if (annotation.target().kind() == AnnotationTarget.Kind.METHOD
                                     && !BUILTIN_HTTP_ANNOTATIONS_TO_METHOD.containsKey(annotation.name())
                                     && !ResteasyReactiveDotNames.PATH.equals(annotation.name())) {
-                                AnnotationValue value = annotation.value();
-                                if (value != null && value.kind() == AnnotationValue.Kind.ARRAY
-                                        && value.componentKind() == AnnotationValue.Kind.NESTED) {
-                                    for (AnnotationInstance annotationInstance : value.asNestedArray()) {
-                                        methodCreator.addAnnotation(annotationInstance);
-                                    }
```

Comment:
@michalszynkiewicz any reason why this was done? I can't think of anything.

Reply:
I think simply doing `methodCreator.addAnnotation` didn't work for group annotations, e.g.:
```
    @ClientHeaderParam(name="OptionalMethodHeader", value="{fail}", required=false)
    @ClientHeaderParam(name="MethodHeaderExplicit", value="SomeValue")
    JsonObject methodOptionalMethodHeaderNotSentWhenComputeThrowsException();

```

Reply:
But that should be tested in the TCK, so if the TCK is passing without it either I was wrong or something got fixed in gizmo

Reply:
Oh I did fix annotation support in Gizmo for sure :-) See https://github.com/quarkusio/gizmo/pull/81

Reply:
I was just wondering how could this possibly have worked, as the previous code simply took container annotations and un-nested one level of them, which IMHO must have resulted in a really weird bytecode... but I guess we have something somewhere to recover from it :-)